dev/src/reference.ts

```diff
@@ -1663,8 +1706,17 @@ export class Query {
         .on('data', result => {
           readTime = result.readTime;
           if (result.document) {
-            const document = result.document;
-            docs.push(document);
+            const document: QueryDocumentSnapshot<T> = result.document;
+            const finalDoc = new DocumentSnapshotBuilder(
+              document.ref.withConverter(this._converter)
+            );
+            // Recreate the QueryDocumentSnapshot with the DocumentReference
+            // containing the original converter.
+            finalDoc.fieldsProto = document.fieldsProto;
```

Comment:
I tried moving it to `_stream()`, but we run into problems since we mock the output of `stream()` in the integration tests, so I think we would be mocking out functionality in that case.

Also, we currently do the conversion for `getAll()` after we receive it in the stream.

Reply:
I was able to get this to work with no changes in the tests: https://gist.github.com/schmidt-sebastian/f021aa6928876ed240a622ca0b3b3aa4. This encapsulates the conversion within the method, which matches the behavior of getAll() (IMHO).